app/services/stat_builder.rb

```diff
@@ -0,0 +1,33 @@
+# typed: true
+# frozen_string_literal: true
+
+# Runs queries and builds the data structure for the summary table.
+class StatBuilder
+  extend T::Sig
+
+  sig { returns(T::Hash[Collection, Hash]) }
+  def self.build_stats
+    collections_by_id = Collection.all.index_by(&:id)
+
+    by_state.each_with_object({}) do |(id, val), obj|
+      obj[collections_by_id.fetch(id)] = val.merge('total' => total.fetch(id))
+    end
+  end
+
+  sig { returns(T::Hash[Integer, T::Hash[String, Integer]]) }
+  def self.by_state
+    Work.all.group(:collection_id, :state).count.each_with_object({}) do |((id, state), count), obj|
```

Comment:
Same question as above re: performance.

Reply:
This doesn't show all, because of the 'group' it just gets one row per collection/status.